internal/controller/nemo_datastorev2_controller.go

```diff
@@ -0,0 +1,642 @@
+	return r.updater
+}
+
+// GetRenderer returns the renderer instance
+func (r *NemoDatastoreV2Reconciler) GetRenderer() render.Renderer {
+	return r.renderer
+}
+
+// GetEventRecorder returns the event recorder
+func (r *NemoDatastoreV2Reconciler) GetEventRecorder() record.EventRecorder {
+	return r.recorder
+}
+
+// GetOrchestratorType returns the container platform type
+func (r *NemoDatastoreV2Reconciler) GetOrchestratorType() (k8sutil.OrchestratorType, error) {
+	if r.orchestratorType == "" {
+		orchestratorType, err := k8sutil.GetOrchestratorType(r.GetClient())
+		if err != nil {
+			return k8sutil.Unknown, fmt.Errorf("Unable to get container orchestrator type, %v", err)
+		}
+		r.orchestratorType = orchestratorType
+		r.GetLogger().Info("Container orchestrator is successfully set", "type", orchestratorType)
+	}
+	return r.orchestratorType, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *NemoDatastoreV2Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.recorder = mgr.GetEventRecorderFor("nemo-datastore-service-controller")
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&appsv1alpha1.NemoDatastoreV2{}).
+		Owns(&appsv1.Deployment{}).
+		Owns(&appsv1.StatefulSet{}).
+		Owns(&corev1.Service{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&networkingv1.Ingress{}).
+		Owns(&autoscalingv2.HorizontalPodAutoscaler{}).
+		WithEventFilter(predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				// Type assert to NemoDatastoreV2
+				if oldNemoDatastoreV2, ok := e.ObjectOld.(*appsv1alpha1.NemoDatastoreV2); ok {
+					newNemoDatastoreV2 := e.ObjectNew.(*appsv1alpha1.NemoDatastoreV2)
+
+					// Handle case where object is marked for deletion
+					if !newNemoDatastoreV2.ObjectMeta.DeletionTimestamp.IsZero() {
+						return true
+					}
+
+					// Handle only spec updates
+					return !reflect.DeepEqual(oldNemoDatastoreV2.Spec, newNemoDatastoreV2.Spec)
+				}
+				// For other types we watch, reconcile them
+				return true
+			},
+		}).
+		Complete(r)
+}
+
+func (r *NemoDatastoreV2Reconciler) refreshMetrics(ctx context.Context) {
+	logger := log.FromContext(ctx)
+	// List all nodes
+	NemoDatastoreV2List := &appsv1alpha1.NemoDatastoreV2List{}
+	err := r.Client.List(ctx, NemoDatastoreV2List, &client.ListOptions{})
+	if err != nil {
+		logger.Error(err, "unable to list NemoDatastoreV2s in the cluster")
+		return
+	}
+	//refreshNemoDatastoreV2Metrics(NemoDatastoreV2List)
+}
+
+func (r *NemoDatastoreV2Reconciler) reconcileNemoDatastoreV2(ctx context.Context, nemoDatastore *appsv1alpha1.NemoDatastoreV2) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	var err error
+	defer func() {
+		if err != nil {
+			r.GetEventRecorder().Eventf(nemoDatastore, corev1.EventTypeWarning, conditions.Failed,
+				"NemoDatastoreV2 %s failed, msg: %s", nemoDatastore.Name, err.Error())
+		}
+	}()
+
+	pvc := corev1.PersistentVolumeClaim{}
```

Comment:
pull this into a separate method `getPVCSpec()`

Reply:
we should consider following fields to be updated in the spec from the `storage` params

```
persistence:
  enabled: true
  create: true
  existingClaim: datastore-shared-storage
  size: 5Gi
  accessModes:
    - ReadWriteOnce
  labels: {}
  annotations: {}
  storageClass:
  subPath:
```